client/go/outline/iptable/stream_dialer.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptable
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+func lookupInTable[D any](table IPTable[D], address string) (foundDialer D, ok bool) {
+	host := address
+	if _host, _, err := net.SplitHostPort(address); err == nil {
+		host = _host
+	}
+
+	ip, err := netip.ParseAddr(host)
+	if err == nil {
+		return table.Lookup(ip)
+	}
+
+	var zeroD D
+	return zeroD, false
+}
+
+// StreamDialer is a [transport.StreamDialer] that routes connections
+// based on the destination IP address using an [IPTable].
+// If a specific route is found in the table, the corresponding dialer is used.
+// Otherwise, the default dialer (if set) is used.
+type StreamDialer struct {
+	table         IPTable[transport.StreamDialer]
+	defaultDialer transport.StreamDialer
```

Comment:
I guess things moved around since my first comment, but I meant this is redundant.